### PR TITLE
Print nice error message if sphinx_changelog is not installed

### PR DESCRIPTION
**Pull request by larrybradley**
Building docs locally requires the`sphinx_changelog` package.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -43,6 +43,12 @@
     print('ERROR: the documentation requires the sphinx-astropy package to be installed')
     sys.exit(1)
 
+try:
+    import sphinx_changelog
```
Comment (pllim) [inline docs/conf.py:47]: Unused import will trigger flake8 when we do make it stricter in the future, so might as well.
```suggestion
    import sphinx_changelog  # noqa
```

**Comment by Cadair** [comment]
Maybe a [more general](https://github.com/sunpy/sunpy/blob/main/docs/conf.py#L12-L25) solution might be nice? :angel:

**Comment by pllim** [comment]
Perhaps... but I'll leave it to Larry whether generalization is out of scope or not.

**Comment by saimn** [comment]
Should sphinx-changelog be added as a dependency for sphinx-astropy ?

**Comment by larrybradley** [comment]
@Cadair Thanks, I implemented a more general solution.

@saimn That seems reasonable to me.

File: docs/conf.py [hunk 2]
```diff
@@ -30,20 +30,32 @@
...
     import importlib.metadata as importlib_metadata
 except ImportError:
     import importlib_metadata
 
-try:
-    from sphinx_astropy.conf.v1 import *  # noqa
-except ImportError:
-    print('ERROR: the documentation requires the sphinx-astropy package to be installed')
+# -- Check for missing dependencies -------------------------------------------
+doc_requires = get_distribution("astropy").requires(extras=("docs",))
```
Comment (pllim) [inline docs/conf.py:42]: Wait a min... I thought we are using importlib now.

https://github.com/astropy/astropy/blob/75075236dc543255e82ea3609ff8ab2449179d46/docs/conf.py#L107

Comment (saimn) [reply docs/conf.py:42]: Yeah indeed, we should use `importlib.metadata` instead (#11091).

Comment (larrybradley) [reply docs/conf.py]: Done.

**Review by saimn — APPROVED** [review]
Sounds good!

**Review by pllim — APPROVED** [review]
LGTM. Thanks!

**Comment by pllim** [comment]
Does this really need backport? I don't think so... I am changing the milestone.

**Pull request merged**